client-src/entities/Tank.ts

```diff
@@ -109,13 +123,13 @@ export class Tank extends MatterContainer {
         return this;
     }
 
-    setUpgrade(upgrade: any) {
-        this.upgrades.push(upgrade)
-        // TODO: update the data base on upgrade type
-        // Example:
-        // this.attackSpeed -= 100;
-        // this.maxHp += 5;
-        // this.hp += 5;
+    setUpgrade(upgrades: UpgradeObject) {
+        Object.keys(upgrades).forEach((key: UpgradeType) => {
+            this[key] += upgrades[key];
+            this.upgrades[key] += upgrades[key];
+        })
+        // always heal at least 1
+        this.hp += Math.max(upgrades.maxHP, 1);
```

Comment:
hp can go over maxHP with this piece of code

should be heal the difference between current maxHP and target maxHP, and then make at least 1